src/rcdgen/check_data.cpp

```diff
@@ -1763,6 +1763,69 @@ static std::shared_ptr<SHOPBlock> ConvertSHOPNode(std::shared_ptr<NodeGroup> ng)
 	return sb;
 }
 
+/**
+ * Convert a node group to a FGTR game block.
+ * @param ng Generic tree of nodes to convert.
+ * @return The created FGTR game block.
+ */
+static std::shared_ptr<FGTRBlock> ConvertFGTRNode(std::shared_ptr<NodeGroup> ng)
+{
+	ExpandNoExpression(ng->exprs, ng->pos, "FGTR");
+	auto block = std::make_shared<FGTRBlock>();
+
+	Values vals("FGTR", ng->pos);
+	vals.PrepareNamedValues(ng->values, true, true);
+
+	const std::string category = vals.GetString("category");
+	if (category == "gentle") block->is_thrill_ride = false;
+	else if (category == "thrill") block->is_thrill_ride = true;
+	else {
+		fprintf(stderr, "Error at %s: Invalid category '%s' (expected 'gentle' or 'thrill').\n", ng->pos.ToString(), category.c_str());
+		return nullptr;
+	}
+	block->tile_width = vals.GetNumber("tile_width");
+	block->ride_width_x = vals.GetNumber("ride_width_x");
+	block->ride_width_y = vals.GetNumber("ride_width_y");
+	block->heights.reset(new int8[block->ride_width_x * block->ride_width_y]);
+	block->ne_views.reset(new std::shared_ptr<SpriteBlock>[block->ride_width_x * block->ride_width_y]);
+	block->se_views.reset(new std::shared_ptr<SpriteBlock>[block->ride_width_x * block->ride_width_y]);
+	block->nw_views.reset(new std::shared_ptr<SpriteBlock>[block->ride_width_x * block->ride_width_y]);
+	block->sw_views.reset(new std::shared_ptr<SpriteBlock>[block->ride_width_x * block->ride_width_y]);
```

Comment:
It doesn't make a difference in performance, but it's useful if you don't copy/paste computations. Make a new variable the contains the size, and use that instead. It makes it easier to see that all arrays are equally large. (Currently, you have to carefully check each line to be exactly the same as the previous line and we are very bad at that, so avoid forcing people into that.)

One of the options for `views` is to make an array of views.
Not sure it's worth the hassle though, as now it's perfectly clear which variable contains which views, something that you loose with `block->views[i]`.